src/toxcore/tcp/server/server.rs

```diff
@@ -375,11 +375,10 @@ impl Server {
             .collect::<Vec<PublicKey>>();
 
         let remove_timedouts = keys.iter()
-            .map(|key| {
-                self.shutdown_client_inner(key, state)
-            });
+            // failure in removing one client should not affect other clients
+            .map(|key| self.shutdown_client_inner(key, state).then(|_| Ok(())));
```

Comment:
What's that for?

Reply:
`shutdown_client_inner` may return failed future if it can't send disconnect notification. In this case we should ignore it and send all other notifications that can be sent.

Reply:
OK, thanks